kafka-connect/src/main/java/io/tabular/iceberg/connect/data/SchemaUtils.java

```diff
@@ -132,6 +138,65 @@ private static boolean typeMatches(org.apache.iceberg.Schema schema, TypeUpdate
     return schema.findType(update.name()).typeId() == update.type().typeId();
   }
 
+  public static PartitionSpec createPartitionSpec(
+      org.apache.iceberg.Schema schema, List<String> partitionBy) {
+    if (partitionBy.isEmpty()) {
+      return PartitionSpec.unpartitioned();
+    }
+
+    PartitionSpec.Builder specBuilder = PartitionSpec.builderFor(schema);
+    partitionBy.forEach(
+        partitionField -> {
+          Matcher matcher = TRANSFORM_REGEX.matcher(partitionField);
```

Comment:
Would it be possible to use the [transforms parsers](https://github.com/apache/iceberg/blob/ede085d0f7529f24acd0c81dd0a43f7bb969b763/api/src/main/java/org/apache/iceberg/transforms/Transforms.java#L42) here instead of regex?

Reply:
I looked into those, but that doesn't deal with the field name, so I don't think it would help much